astropy/table/table.py

```diff
@@ -842,8 +842,8 @@ def __getitem__(self, item):
             return self.columns[item]
         elif isinstance(item, (int, np.integer)):
             return self.Row(self, item)
-        elif isinstance(item, (tuple, list)) and all(isinstance(x, six.string_types)
-                                                     for x in item):
+        elif (isinstance(item, (tuple, list)) and item and
```

Comment:
Half the battle of writing better code is knowing what will make people complain.